x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceConfigUpdate.java

```diff
@@ -6,14 +6,48 @@
 package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
 
 import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
+import org.elasticsearch.xpack.core.ml.inference.results.WarningInferenceResults;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
 import org.elasticsearch.xpack.core.ml.utils.NamedXContentObject;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 
 public interface InferenceConfigUpdate extends NamedXContentObject, NamedWriteable {
+    Set<String> RESERVED_ML_FIELD_NAMES = new HashSet<>(Arrays.asList(
+        WarningInferenceResults.WARNING.getPreferredName(),
+        TrainedModelConfig.MODEL_ID.getPreferredName()));
 
     InferenceConfig apply(InferenceConfig originalConfig);
 
     InferenceConfig toConfig();
 
     boolean isSupported(InferenceConfig config);
+
+    String getResultsField();
+
+    InferenceConfigUpdate duplicateWithResultsField(String resultsField);
```

Comment:
I know it will mean more LoC, but this method is not extensible and only exists for a single thing/path. 

```suggestion
    Builder newBuilder();

    interface Builder<T extends Builder, U extends InferenceConfigUpdate> {

        U build();

        T setResultsField(String resultsField);

    }
```
Then each builder satisfies this interface and needs a copy ctor accepting their appropriate `update` object.

The code to replace the resultsField name is now
```
InferenceConfigUpdate newUpdate = update.newBuilder().setResultsField("foo").build();
```

This seems more general + more clean than a one off method.

Reply:
Good suggestion I've implemented the builder